sources/driverdog/src/main.rs

```diff
@@ -276,9 +311,49 @@ where
     Ok(())
 }
 
+// Loads the kernel modules for the given configuration, and for the given kernel version
+fn copy_modules<S>(driver_config: &CopyingDriverConfig, kernel_version: S) -> Result<()>
+where
+    S: AsRef<str>,
+{
+    let kernel_version = kernel_version.as_ref();
+
+    // Destination for the kernel modules
+    let modules_path = Path::new(LIB_MODULES_PATH)
+        .join(kernel_version)
+        .join(&driver_config.lib_modules_path);
+
+    // Next, copy the kernel modules
+    for (name, module) in driver_config.kernel_modules.iter() {
+        copy_kernel_module(name, &modules_path, &module.copy_source)?;
+    }
+
+    Ok(())
+}
+
+// Copy the module to the modules path provided
+fn copy_kernel_module<S, P>(name: S, modules_path: P, driver_source_path: P) -> Result<()>
```

Comment:
nit: It's silly but these `module_path` and `driver_source_path` can't be arbitrary and different "path-like" types unless you do this.

```suggestion
fn copy_kernel_module<S, P1, P2>(name: S, modules_path: P1, driver_source_path: P2) -> Result<()>
```